### PR TITLE
[#IC-383] Add notice_number to message_view

### PR DESCRIPTION
**Pull request by fabriziopapi**
Add notice_number from payment message content to mesage_view.

#### List of Changes

add notice_number

#### Motivation and Context

do not retrieve message content from blob to enrich message metadata 

#### How Has This Been Tested?

unit test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/models/message_view.ts [hunk 1]
```diff
@@ -8,20 +8,31 @@ import { ServiceId } from "../../generated/definitions/ServiceId";
...
 
+export const PaymentComponent = t.union([
+  NotHasComponent,
+  t.intersection([
+    HasComponent,
+    t.interface({ notice_number: PaymentNoticeNumber })
+  ])
+]);
+export type PaymentComponent = t.TypeOf<typeof PaymentComponent>;
+
```
Comment (gquadrati) [inline src/models/message_view.ts:30]: Please, add a unit test

Comment (fabriziopapi) [reply src/models/message_view.ts:30]: added! Thanks.

File: src/models/__tests__/message_view.test.ts [hunk 2]
```diff
@@ -66,6 +66,51 @@ const containerMock = ({
...
+    const messageViewWithPayment = {
+      ...aMessageView,
+      components: {
+        ...aMessageView.components,
+        payment: { has: false, notice_number: aNoticeNumber }
+      }
+    };
+    const result = MessageView.decode(messageViewWithPayment);
+    expect(E.isRight(result)).toBeTruthy();
+  });
```
Comment (gquadrati) [inline src/models/__tests__/message_view.test.ts:111]: Title says that we are expecting this decode to fail

Comment (fabriziopapi) [reply src/models/__tests__/message_view.test.ts:111]: fixed, thank you!

**Review by gquadrati — APPROVED** [review]
LGTM

**Review by AleDore — APPROVED** [review]

**Pull request merged**